### ISSUE TITLE
RGI requires writable file system

### ISSUE THREAD

**Issue by caspargross** [arpcard/rgi#60]
Hello, 
I want to use rgi inside a Singularity container. Unfortunately rgi requires write access to its install path for execution, even when the database was already loaded during container creation.  It exits with this error message:  
````
Traceback (most recent call last):
  File "/opt/conda/envs/PI_env/bin/rgi", line 4, in <module>
    MainBase()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/MainBase.py", line 54, in __init__
    getattr(self, args.command)()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/MainBase.py", line 59, in main
    self.main_run(args)
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/MainBase.py", line 91, in main_run
    rgi_obj.run()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/RGI.py", line 137, in run
    self.create_databases()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/RGI.py", line 131, in create_databases
    db_obj.build_databases()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/Database.py", line 22, in build_databases
    self.write_fasta_from_json()
  File "/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/Database.py", line 68, in write_fasta_from_json
    with open(os.path.join(self.db, "proteindb.fsa"), 'w') as fout:
OSError: [Errno 30] Read-only file system: '/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/_db/proteindb.fsa'
```` 
 Is there a way to run rgi inside a singularity container (without the need of overlay and root access?)
Maybe it is posible to provide a writable path via a parameter where rgi can store its files?

**Comment by raphenya** [comment]
@caspargross Are you able to create an environment on the singularity container? If you can, then try using the environment to install. Note that RGI runs successfully on python version 3.6 for both Linux and Mac OS.

**Comment by caspargross** [comment]
Thanks for your fast reply. RGI is installed in a conda environment inside the container. 

I found a hacky solution that kind of works. I created an empty folder  and bind it to the `_db` folder inside the  container with the following singularity run parameters:
```
-B /path/to/local_db:/opt/conda/envs/PI_env/lib/python3.5/site-packages/app/_db
```

Maybe it would be a good option to create a separate, optionally user defined folder for all files written during the analysis anyway?
